src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -133,7 +134,8 @@
         (when editing?
           [comps/SidebarButton {:color :exception-state :margin :top
                                 :text "Cancel Editing" :icon :cancel
-                                :onClick #(stop-editing state)}])]))])
+                                :onClick #(do (react/call :load-new-method-template this (:prev-snapshot-id @state))
```

Comment:
Thanks, good tips. I'll probably just pass the method directly instead of `this`, as you suggested below.

Re (3), I initially tried that and it didn't work: when I click Cancel it calls `:load-new-method-template` with the _current_ snapshot-id, not the original snapshot-id before edit was clicked. I could be missing something, but I think the `render-side-bar` function gets called while in edit mode, which overwrites the `onClick` callback on the Cancel button. Saving the previous snapshot-id in `@state` works around that.